crates/node-core/src/init.rs

```diff
@@ -391,6 +392,12 @@ pub fn init_from_state_dump<DB: Database>(
         );
     }
 
+    // insert sync stages for stages that require state
+    let tx = provider_rw.tx_ref();
+    for stage in StageId::STATE_REQUIRED.iter() {
+        tx.put::<tables::StageCheckpoints>(stage.to_string(), StageCheckpoint::new(block))?;
+    }
```

Comment:
is it enough to add the block number to stage checkpoints here? @Rjected

Reply:
We need to change the above code that fetches the latest block and passes it to the `insert_state` etc. also style nit: we should use the `save_stage_checkpoints` instead of using the raw tx ref

Reply:
> etc.
do you just want to push to this branch?
> save_stage_checkpoints
didn't find this in the code :( same q to you here

Reply:
here I was rather referring to: do we need to set `StageUnitCheckpoint` too?

Reply:
Ah, I think for these stages since they're complete, we can use `None` instead of explicitly providing a `StageUnitCheckpoint`.

The method I was referring to is this method on `provider_rw`:
https://github.com/paradigmxyz/reth/blob/9e7cf69413f44d02606cc2250d4b77397150a165/crates/storage/provider/src/traits/stage_checkpoint.rs#L21